netty-utils/src/test/java/org/asynchttpclient/netty/util/ByteBufUtilsTests.java

```diff
@@ -0,0 +1,67 @@
+package org.asynchttpclient.netty.util;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.powermock.api.mockito.PowerMockito;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.internal.junit.ArrayAsserts;
+import java.nio.charset.Charset;
+
+public class ByteBufUtilsTests {
+
+    @Test
+    public void testByteBuf2Bytes() {
+        ByteBuf byteBuf = Unpooled.copiedBuffer(new byte[]{'f', 'o', 'o'});
+        ArrayAsserts.assertArrayEquals(ByteBufUtils.byteBuf2Bytes(byteBuf),
+                new byte[]{'f', 'o', 'o'});
+
+        ByteBuf buf = PowerMockito.mock(ByteBuf.class);
```

Comment:
Would `Unpooled.EMPTY_BUFFER` work here, maybe?

Reply:
Nope.. It didn't :( as an array is being created

Reply:
Done.. used `Unpooled.buffer();` instead of mocking :)